### ISSUE TITLE
DevTools broken on latest Dart with ddk-related issue.

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/devtools#1095]
Example failure: https://travis-ci.org/flutter/devtools/jobs/590135205
@jacob314

**Comment by kenzieschmoll** [comment]
After some further investigation, I think this test is related to ddk. The ddc bots are broken: https://travis-ci.org/flutter/devtools/builds/590186628?utm_source=github_status&utm_medium=notification

DevTools is broken on the latest dart 2.6.0-dev.3.0 (could be broken on older versions, but I can confirm that it is not broken on dart stable 2.5.0). It is broken with the ddk build, but is WAI with the dart2js build (both built from source and ran from pub).

Here are examples of the errors we are seeing: 
```
Uncaught (in promise) Error: NoSuchMethodError: '<Unexpected Null Value>'
method not found
Receiver: null
Arguments: []
    at Object.throw_ [as throw] (dart_sdk.js:4592)
    at Object.throwNullValueError (dart_sdk.js:4545)
    at Object._notNull [as notNull] (dart_sdk.js:4896)
    at Duration.get inMilliseconds [as inMilliseconds] (dart_sdk.js:40408)
    at Function._createTimer (dart_sdk.js:28026)
    at _rootCreateTimer (dart_sdk.js:28659)
    at async._CustomZone.new.createTimer (dart_sdk.js:28492)
    at Function.new (dart_sdk.js:28013)
    at Object.waitForDimensionsComputed (analytics_platform.dart:80)
    at setupAndGaScreen (analytics_platform.dart:99)
    at setupAndGaScreen.next (<anonymous>)
    at runBody (dart_sdk.js:22603)
    at Object._async [as async] (dart_sdk.js:22631)
    at Object.setupAndGaScreen (analytics_platform.dart:95)
    at html_framework.dart:265
    at _rootRun (dart_sdk.js:28605)
    at async._CustomZone.new.run (dart_sdk.js:28434)
    at _FutureListener.whenComplete.handleWhenComplete (dart_sdk.js:24526)
    at handleWhenCompleteCallback (dart_sdk.js:24981)
    at Function._propagateToListeners (dart_sdk.js:25038)
    at _Future.new.[_completeWithValue] (dart_sdk.js:24901)
    at dart_sdk.js:24919
    at _rootRun (dart_sdk.js:28608)
    at async._CustomZone.new.run (dart_sdk.js:28434)
    at async._AsyncCallbackEntry.new.callback (dart_sdk.js:28378)
    at Object._microtaskLoop (dart_sdk.js:25256)
    at _startMicrotaskLoop (dart_sdk.js:25262)
    at dart_sdk.js:25337
```
This error goes away if we remove `const` from creation of `Duration()`. This seems like a recent regression.

Another error that may be ddk related is showing up for the inspector page:
```
Uncaught (in promise) Error: Failed assertion: boolean expression must not be null
    at Object.throw_ [as throw] (dart_sdk.js:4592)
    at Object._throwBooleanConversionError (dart_sdk.js:4882)
    at Object.test (dart_sdk.js:4874)
    at Latin1Codec.decode (dart_sdk.js:36906)
    at response$.Response.bytes.get body [as body] (response.dart:27)
    at _loadHelper (flutter_widget.dart:116)
    at _loadHelper.next (<anonymous>)
    at onValue (dart_sdk.js:22584)
    at _rootRunUnary (dart_sdk.js:28614)
    at async._CustomZone.new.runUnary (dart_sdk.js:28441)
    at _FutureListener.thenAwait.handleValue (dart_sdk.js:24508)
    at handleValueCallback (dart_sdk.js:25009)
    at Function._propagateToListeners (dart_sdk.js:25041)
    at _Future.new.[_completeWithValue] (dart_sdk.js:24901)
    at dart_sdk.js:24919
    at _rootRun (dart_sdk.js:28608)
    at async._CustomZone.new.run (dart_sdk.js:28434)
    at async._AsyncCallbackEntry.new.callback (dart_sdk.js:28378)
    at Object._microtaskLoop (dart_sdk.js:25256)
    at _startMicrotaskLoop (dart_sdk.js:25262)
    at dart_sdk.js:25337
```
I have stepped through this code, and it looks like it is related to some undefined values in `dart_sdk.js`?

**Comment by kenzieschmoll** [comment]
@markzipan

**Comment by Markzipan** [comment]
It's possible this has something to do with Vijay's update to private names for const objects. It seems private dart-sdk objects (like Duration) are having trouble accessing their own members.

@vsmenon, see: https://dart-review.googlesource.com/c/sdk/+/118261

**Comment by kenzieschmoll** [comment]
Just did some version checking. Looks like the regression started in the 2.6.0-dev.1.0. I do not see the issue in 2.6.0-dev.0.0.

**Comment by devoncarew** [comment]
I've temporarily pegged us to `2.6.0-dev.0.0` (https://github.com/flutter/devtools/pull/1102/files), but we should revert that as soon as there's a fix available.

**Comment by vsmenon** [comment]
Do we know if this is broken wrt master?

Over the past couple weeks, I checked in a problematic fix (to a different issue), rolled it back, and re-landed.  You may have caught the buggy version in between, or there may still be an issue on the latest.

**Comment by kenzieschmoll** [comment]
I am still seeing this at master

**Comment by Markzipan** [comment]
I'm currently debugging through this. It seems to be something less obvious than cross-module, cross-dart-library static consts (which has popped up before).

**Comment by Markzipan** [comment]
I've bisected the breaking commit between a6cc7b94c7b6d1b8441f2813ba724f6d0550ca97 and acac9ab11b25abb490dfd0b632f570a01772df79. This seems to strongly implicate the re-land that @vsmenon mentioned above. I can't seem to get a local cross-module repro, however.

**Comment by Markzipan** [comment]
This is probably the same issue as: https://github.com/dart-lang/build/issues/2483

**Comment by vsmenon** [comment]
@kenzieschmoll - can you please try a `pub upgrade` and see if the problem persists?  The latest `build_web_compilers` should fix this issue.

**Comment by kenzieschmoll** [comment]
Looks like the problem is fixed with build_web_compilers 2.6.2. thanks!
